ci/nightly/coverage.py

```diff
@@ -19,48 +19,71 @@
 pipeline.template.yml and the docstring on `trim_pipeline` below.
 """
 
+import subprocess
 import sys
 from pathlib import Path
 
-import yaml
-
 import materialize.cli.mzcompose
 
 
 def main() -> int:
-    with open(Path(__file__).parent.parent / "test" / "pipeline.template.yml") as f:
-        pipeline = yaml.safe_load(f)
-
-    tests = []
-
-    for step in pipeline["steps"]:
-        for plugin in step.get("plugins", []):
-            for plugin_name, plugin_config in plugin.items():
-                if plugin_name == "./ci/plugins/mzcompose":
-                    tests.append((plugin_config["composition"], plugin_config["run"]))
+    # Just the fast sqllogictests, for now
+    tests = [("sqllogictest", "default")]
 
     for (composition, workflow) in tests:
-        print(f"==> Running workflow {workflow} in {composition}")
         materialize.cli.mzcompose.main(
             [
-                "run",
-                workflow,
-                "--coverage",
                 "--find",
                 composition,
+                "--coverage",
+                "run",
+                workflow,
             ]
         )
         materialize.cli.mzcompose.main(
             [
-                "down",
-                "-v",
-                "--coverage",
                 "--find",
                 composition,
+                "down",
+                "-v",
             ]
         )
 
-    # TODO: gather and combine coverage information.
+    # NB: mzcompose _munge_services() sets LLVM_PROFILE_FILE
+    subprocess.run(
+        [
+            "rust-profdata",
+            "merge",
+            "-sparse",
+            *Path("test/sqllogictest/coverage").glob("*.profraw"),
+            "-o",
+            "sqllogictest.profdata",
+        ]
+    )
+    with open("coverage-sqllogictest.lcov", "w") as out:
+        subprocess.run(
+            [
+                "rust-cov",
+                "export",
+                "--format=lcov",
```

Comment:
Any reason to export as lcov?

Reply:
I'm in the middle of experimenting with a bunch of ways of presenting the output.  Originally I exported the JSON, which no tools seem to consume at all... so at least lcov output can be consumed by lcov's genhtml.  Which I know is more useful than llvm-cov show's own HTML format which is completely unusable because it produces a giant, multi-hundred-megabyte HTML file.  Suggestions appreciated though!